### PR TITLE
Bulk load CDK: test refactors

### PR DESCRIPTION
**Pull request by edgao**
just look at the commits individually, they're hopefully organized in a reasonable way.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-s3-v2/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+testExecutionConcurrency=-1
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-s3-v2/gradle.properties:1]: I don't understand what changed, but this used to not be necessary https://airbytehq-team.slack.com/archives/C06E6CQMDT3/p1731520778573749

File: airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/util/Jsons.kt [hunk 2]
```diff
@@ -19,13 +20,19 @@ import java.math.BigInteger
 import java.nio.ByteBuffer
 
 object Jsons : ObjectMapper() {
+    // allow jackson to deserialize anything under 100 MiB
+    // (the default, at time of writing 2024-05-29, with jackson 2.15.2, is 20 MiB)
+    private const val JSON_MAX_LENGTH = 100 * 1024 * 1024
```
Comment (edgao) [inline airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/util/Jsons.kt:25]: this diff is just matching our existing behavior from https://github.com/airbytehq/airbyte/blob/712ea82d050fd731b3289697bc56c214a3a33955/airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/json/Jsons.kt#L34

(it's also needed to make the float-handling test work)

File: airbyte-cdk/bulk/toolkits/load-parquet/src/main/kotlin/io/airbyte/cdk/load/file/parquet/ParquetWriter.kt [hunk 3]
```diff
@@ -52,7 +53,8 @@ fun OutputStream.toParquetWriter(
     val writer =
         AvroParquetWriter.builder<GenericRecord>(outputFile)
             .withSchema(avroSchema)
-            .withConf(Configuration())
+            // needed so that we can have arrays containing null elements
+            .withConf(Configuration().apply { setBoolean(WRITE_OLD_LIST_STRUCTURE, false) })
```
Comment (edgao) [inline airbyte-cdk/bulk/toolkits/load-parquet/src/main/kotlin/io/airbyte/cdk/load/file/parquet/ParquetWriter.kt:57]: again, just matching existing behavior. https://github.com/airbytehq/airbyte/pull/23788/files#diff-0841c4748c4a6dd2d0f2140c21a9c4d4c4deba9e6ed025a868a8284003c1d015

File: airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/IntegrationTest.kt [hunk 4]
```diff
@@ -45,6 +45,10 @@ abstract class IntegrationTest(
     val destinationCleaner: DestinationCleaner,
     val recordMangler: ExpectedRecordMapper = NoopExpectedRecordMapper,
     val nameMapper: NameMapper = NoopNameMapper,
+    /**
+     * See [RecordDiffer.nullEqualsUnset].
+     */
+    val nullEqualsUnset: Boolean = false,
```
Comment (johnny-schmidt) [inline airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/IntegrationTest.kt:51]: Thanks! This one was a giant pain

File: airbyte-cdk/bulk/toolkits/load-parquet/src/main/kotlin/io/airbyte/cdk/load/file/parquet/ParquetWriter.kt (hunk 3 above)
Comment (johnny-schmidt) [reply airbyte-cdk/bulk/toolkits/load-parquet/src/main/kotlin/io/airbyte/cdk/load/file/parquet/ParquetWriter.kt:57]: 😊

File: airbyte-integrations/connectors/destination-s3-v2/gradle.properties (hunk 1 above)
Comment (benmoriceau) [reply airbyte-integrations/connectors/destination-s3-v2/gradle.properties:1]: Did we upgrade gradle?

**Review by benmoriceau — APPROVED** [review]

File: airbyte-integrations/connectors/destination-s3-v2/gradle.properties (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-s3-v2/gradle.properties:1]: :shrug: I didn't bother digging into the commit history for this, the workaround is easy enough

**Review by edgao — COMMENTED** [review]
lgtm (can't approve b/c I opened the pr 😓 )

**Pull request merged**